projects/com.oracle.truffle.llvm.runtime/src/com/oracle/truffle/llvm/runtime/floating/LLVM80BitFloat.java

```diff
@@ -448,16 +450,39 @@ public int hashCode() {
         return bb.array();
     }
 
-    public static LLVM80BitFloat fromBytes(byte[] bytes) {
+    public byte[] getBytes() {
+        ByteBuffer bb = ByteBuffer.allocate(BYTE_WIDTH);
+        bb.order(ByteOrder.LITTLE_ENDIAN);
```

Comment:
Do we really need `LITTLE_ENDIAN` here? If we don't care, I think it's better to have the platform endianness, since it's faster.

Reply:
If this code is run on a big endian machine, the following `putShort`/`putLong` have to be swapped. Since the whole data type is called `X86_FP80` it's an x86 specific data type that should never exist in non-x86 bitcode (therefore it should never exist in bitcode that runs on a big endian machine).

Reply:
Good point. We might at some point want to be able to run foreign bitcode, but not yet, and anyway in that scenario we will have tons of endian conversions all over the place anyway, so this one is probably not going to hurt much...